explore/realspace.py

```diff
@@ -105,23 +110,25 @@ def volume(self):
         raise NotImplementedError()
 
     def sample(self, density):
-        # type: (float) -> np.ndarray[N], np.ndarray[N, 3]
+        # type: (float) -> Tuple[np.ndarray[N], np.ndarray[N, 3]]
         raise NotImplementedError()
 
     def dims(self):
-        # type: () -> float, float, float
+        # type: () -> Tuple[float, float, float]
         raise NotImplementedError()
 
     def rotate(self, theta, phi, psi):
```

Comment:
What coordinate system are these defined in?

Reply:
dims is the shape bounding box, distance units unspecified.

The rotation angles are a more complicated question since I'm sometimes using Euler angles and other times using Tait-Bryan pseudo-nautical units with (ψ,θ,φ) = (roll, pitch, yaw) around shape axes (c, a, b) corresponding to lab coordinates (z, x, y) with z in the direction of the beam and (x,y) the detector (horizontal, vertical). I didn't write this down in the code since I'm not certain it is correct. I suppose I can add it with a disclaimer.

Reply:
I always worry when I see Euler angles etc. So easy to get them wrong.

Mainly though, I was fishing for a comment that makes it clear which rotation is applied first (i.e. is it in the "world" or "object" frame)

Reply:
Yes. I wrote `sasmodels/jitter.py` in order to get it right.

It turns out that my simulation is rotating q space rather than rotating the shape so the rotation code wasn't getting exercised. Which also explains why it can get away with using the "wrong" orientation definition.

There are two orientation systems in sasmodels, one for orienting the shape and another for applying shape distributions. You can see this by running `python -m sasmodels.jitter` setting Δθ=6° and Δφ=10° (less than 5° is treated as zero). This shows the mesh of orientations that are sampled in the integral average. Adjust θ, φ and ψ to show that the mesh is the same regardless of particle orientation. The Euler system of yaw, pitch and roll is better for user control: φ rotates the pattern around in the plane, θ moves the c-axis into the plane, ψ rolls the shape on its axis. Much easier than the previous system of Tait-Bryan coordinates.

The particle tumbling is easier to understand using nautical angles. It isn't ideal since there is pinching at the poles. Compare (large Δθ, small Δφ) to (small Δθ, large Δφ) in `python -m sasmodels.jitter`.  There are map projections that handle this better but we opted for simplicity.